tg_bot/modules/antiflood.py

```diff
@@ -0,0 +1,183 @@
+                sql.set_flood(chat.id, 0)
+                message.reply_text("L'antiflood è stato disabilitato.")
+                return (
+                    "<b>{}:</b>"
+                    "\n#SETFLOOD"
+                    "\n<b>Admin:</b> {}"
+                    "\nDisabilitazione antiflood.".format(
+                        html.escape(chat.title), mention_html(user.id, user.first_name)
+                    )
+                )
+
+            elif amount < 3:
+                message.reply_text(
+                    "L'antiflood deve essere impostato a 0 (disabilitato), o un numero maggiore di 3!"
+                )
+                return ""
+
+            else:
+                sql.set_flood(chat.id, amount)
+                message.reply_text(
+                    "Modulo antiflood aggiornato e impostato a {}".format(amount)
+                )
+                return (
+                    "<b>{}:</b>"
+                    "\n#SETFLOOD"
+                    "\n<b>Admin:</b> {}"
+                    "\nAntiflood impostato a <code>{}</code>.".format(
+                        html.escape(chat.title),
+                        mention_html(user.id, user.first_name),
+                        amount,
+                    )
+                )
+
+        else:
+            message.reply_text(
+                "Argomento non riconosciuto - usare un numero, 'off', oppure 'on'."
+            )
+
+    return ""
+
+
+@run_async
+def flood(bot: Bot, update: Update):
+    chat = update.effective_chat  # type: Optional[Chat]
+
+    limit = sql.get_flood_limit(chat.id)
+    if limit == 0:
+        update.effective_message.reply_text(
+            "Sto monitorando la chat alla ricerca di flooding!"
+        )
+    else:
+        update.effective_message.reply_text(
+            "Bannerò gli utenti che invieranno più di {} messaggi consecutivi.".format(
+                limit
+            )
+        )
+
+
+def __migrate__(old_chat_id, new_chat_id):
+    sql.migrate_chat(old_chat_id, new_chat_id)
+
+
+def __chat_settings__(chat_id, user_id):
+    limit = sql.get_flood_limit(chat_id)
+    if limit == 0:
+        return "*Non* sto controllando la chat dai flood."
+    else:
+        return "Antiflood impostato a  `{}` messaggi.".format(limit)
+
+
+__help__ = """
+ - /flood: Restituisce le attuali impostazioni di flood
+
+*Admin only:*
+ - /setflood <int/'on'/'off'>: Abilita o disabilita l'antiflood
+ - /setmsglen <int/'on'/'off'>: Abilita o disabilita l'antiflood sulla lunghezza del messaggio
+"""
+
+__mod_name__ = "AntiFlood"
+
+FLOOD_BAN_HANDLER = MessageHandler(
+    Filters.all & ~Filters.status_update & Filters.group, check_flood
+)
+SET_FLOOD_HANDLER = CommandHandler(
+    "setflood", set_flood, pass_args=True, filters=Filters.group
+)
+FLOOD_HANDLER = CommandHandler("flood", flood, filters=Filters.group)
+
+dispatcher.add_handler(FLOOD_BAN_HANDLER, FLOOD_GROUP)
+dispatcher.add_handler(SET_FLOOD_HANDLER)
+dispatcher.add_handler(FLOOD_HANDLER)
```

Comment:
Commento qui alla fine del file perché riguarda tutto il modulo ma non volevo mettere tutto il codice. Dovremmo organizzarci per testare questo modulo. In particolare sembra che alcuni client telegram non supportino la modalità lenta, quindi dovremmo provare con quelli.

Reply:
Penso che quello che utilizzi di più client "custom" sia @hearot o sbaglio?